internal/hns/hnsendpoint.go

```diff
@@ -10,6 +10,28 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// EndpointState represents the states of an HNS Endpoint lifecycle.
+type EndpointState uint16
+
+// EndpointState const
+// The lifecycle of an Endpoint goes through created, attached, AttachedSharing - endpoint is being shared with other containers,
+// detached, after being attached, degraded and finally destroyed.
+// Note: This attribute is used by calico to define stale containers and is dependent on HNS v1 api, if we move to HNS v2 api we will need
```

Comment:
@sbangari @ritikaguptams  Is there a way to add a unit test that fails when we moved to v2? I think this would be really important to have to avoid any regression in the future.

Reply:
Is it that only Calico that is using the v1 endpoint?  There are tests in https://github.com/microsoft/hcsshim/blob/main/hcn/hcnendpoint_test.go  but I think that is v2 api?

Reply:
Hi @kiashok,
 
My understanding is hcsshim uses exposes both v1 and v2 apis via the hcsshim/hcn code, which is documented and has testing support. Our codeflow currently is a minor change to add an attribute in an v1 api call. Underlying understanding is calico currently runs on HNS v1 support. We should not have any breaking changes from this change as users on v2 api call would use [Hcshim/hcn/hcnendpoint ListEndpointsofNetwork](https://github.com/microsoft/hcsshim/blob/75428d123ada3f341e1c55e13e738a92d11cbfd6/hcn/hcnendpoint.go#L265C6-L265C28) instead of [ Hcshim/hnsendpoint HNSListEndpointRequest](https://github.com/microsoft/hcsshim/blob/75428d123ada3f341e1c55e13e738a92d11cbfd6/hnsendpoint.go#L41) 

The comments in there are only for documentation purpose i.e. if ever calico moves to v2 they would have to redo their logic on stale endpoints, which uses 'State' as that attribute is not exposed via HNS v2 api currently.

Reply:
sounds good, thanks for the detailed response. We should open an issue on Calico project so they know they are using HNS v1 api's in some places.

Reply:
Will do!